### ISSUE TITLE
fatal error: Unsupported: typedef changes linkage of anonymous type in TPSStripe

### ISSUE THREAD

**Issue by brianhiss** [tipsi/tipsi-stripe#131]
### Issue

When I add in a pod that has a dependency on the gRPC pod, there are fatal build errors:

`./ios/Pods/gRPC-Core/include/grpc/impl/codegen/gpr_types.h:43:3: Unsupported: typedef changes linkage of anonymous type, but linkage was already computed`

It seems that adding `pod 'Firebase/Firestore'` brings with it several additional dependencies like `gRPC` that cause conflicts with `tipsi-stripe`.

Removing `react-native unlink tipsi-stripe` will allow me to compile with `pod 'Firebase/Firestore'`.

You can replicate this issue in a fresh install easily:
- `react-native init TestProject`
- `cd TestProject`
- `npm install`
- `npm install --save tipsi-stripe`
- `react-native link tipsi-stripe`

Edit your `Podfile` to look like:
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'TestProject' do
  pod 'Stripe', '~> 10.1.0'
  inherit! :search_paths

  pod 'Firebase/Core'
  pod 'Firebase/Firestore'
end
```
Run `pod update`:

![screen shot 2017-10-08 at 9 56 56 am](https://user-images.githubusercontent.com/1640257/31314672-0f935cc0-ac0f-11e7-9bed-42438db75066.png)

Compile Errors:
<img width="1115" alt="screen shot 2017-10-07 at 7 07 17 am" src="https://user-images.githubusercontent.com/1640257/31305869-3a4cb074-ab4c-11e7-9d25-837d806e0263.png">

### Environment

1. Application Target Platform: iOS 11
2. Development Operating System: macOS Sierra 10.12.6 (16G29)
3. Build Tools: Xcode 9.0
4. React Native version: 0.49.2
5. tipsi-stripe Version: 3.7.1

**Comment by isnifer** [comment]
@brianhiss not sure, but maybe ver 4 will help you. Currently, I do not have any idea

**Comment by isnifer** [comment]
In ver 4 we will upgrade Stripe pod

**Comment by brianhiss** [comment]
@isnifer When is Version 4 coming out? Is there a branch up here, I can't find one. I'll post in the Google forums as well. Thanks for the quick response!

**Comment by isnifer** [comment]
@brianhiss I hope you'll find it here - https://github.com/tipsi/tipsi-stripe/pulls

**Comment by brianhiss** [comment]
I was able to find the conflict.

It seems that you're allowing all the headers inside the `Pods` directory, `${SRCROOT}/../../../ios/Pods/Headers/Public/**`.

This can potentially create conflicts in the global space. I was able to fix that by only looking in the `Stripe` pod, `${SRCROOT}/../../../ios/Pods/Headers/Public/Stripe/**`. 

This did create issues in finding `<Stripe/Stripe.h>` (https://github.com/tipsi/tipsi-stripe/issues/116), but I was able to edit `ios/TPSStripe/TPSStripeManager.h` and `ios/TPSStripe/TPSCardField.h` to `"Stripe.h"`.
https://github.com/brianhiss/tipsi-stripe/commit/aa23d67cafbda1878b9500950c8e67297c1bfcf9

I can create a pull request if you want to bring in these changes. 


![screen shot 2017-10-09 at 8 37 51 am](https://user-images.githubusercontent.com/1640257/31333951-b923dfc2-acf4-11e7-9283-27a6a741f39b.png)

![screen shot 2017-10-09 at 9 10 48 am](https://user-images.githubusercontent.com/1640257/31333957-bcf05770-acf4-11e7-999d-d9fc47e02f71.png)

**Comment by isnifer** [comment]
Fixed in #141
